src/components/TodoList/TodoList.tsx

```diff
@@ -1,6 +1,18 @@
 import React from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
 
-export const TodoList: React.FC = () => (
+type Props = {
+  todos: Todo[];
+  selectedTodoId?: number;
+  onSelect?: (todo: Todo) => void;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectedTodoId,
+  onSelect = () => {},
```

Comment:
You are initializing your state with props. It can be problematic as it leads to bugs and inconsistencies in the state. You can read more about this [here](https://stackoverflow.com/a/50403930), [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).

Reply:
how am I supposed to do this without using props?